src/device-registry/utils/create-activity.js

```diff
@@ -261,6 +262,16 @@ const createActivity = {
               3
             ),
           };
+
+          const responseFromCreateApproximateCoordinates = distance.createApproximateCoordinates(
+            { latitude, longitude }
+          );
```

Comment:
When creating an activity like deployment, also generate approximate coordinates using the provided ones and save them separately in the corresponding new fields in Site details..